openml/config.py

```diff
@@ -28,7 +28,8 @@
 
 # Default values are actually added here in the _setup() function which is
 # called at the end of this module
-server = _defaults['server']
+server = str(_defaults['server'])  # so mypy knows it is a string
```

Comment:
you could also use `mypy.cast` to not have to use a comment here

Reply:
I generally prefer this for simple types, as it is clear what is happening (whereas you have to be familiar with `mypy.cast` to immediately know what's being done and why).